articles/hapi-vs-koa-vs-express/index.md

```diff
@@ -0,0 +1,159 @@
+Lets see a code example to illustrate this.
+
+```JavaScript
+const express = require('express')
+const app = express()
+
+app.listen(2400, () => {console.log('Server started at port 2400')})
+```
+As we mentioned earlier, Koa uses the same logic as Express. We import the Koa module, instantiate and store it in a variable and then use its listen method to start the server.
+The code snippet below shows how we start a server using Koa.js.
+
+```JavaScript
+const koa = require('koa')
+const app = koa()
+
+app.listen(2400, () => {console.log('Server started at port 2400')})
+```
+
+Now let's look at how Hapi stands out. In Hapi, we import the hapi module, then, instead of instantiating the module, we use its server method to create a server and then use its connection method to connect the server to a specific port. Then we use the start method to start the server.
+Lets see a code example.
+
+```JavaScript
+const hapi = require('hapi')
+const server = new hapi.Server()
+
+server.connection({
+    host: 'localhost',
+    port: '2400'
+})
+
+server.start(error => {
+    if (error) throw error
+    else console.log('Server running at PORT 2400');
+})
+```
+
+### Framework advantages and disadvantages
+In this section we are going to highlight some of the advantages and disadvantages of each framework.
+
+#### Hapi.js
+##### Advantages
+1. Hapi enables one to build a scalable web application or API.
+2. Hapi uses plugins which helps in scaling.
+3. Hapi can also be used in conjunction with front-end frameworks like Vue and React to build single-page web applications.
+4. Hapi provides a deeper control over request handling through the use of context.
+5. Hapi provides input/output validation, routing and caching, hence making it a good choice for building REST APIs.
+6. Hapi provides good plugins such as Yar, Good and Catbox etc.
+
+##### Disadvantages
+1. Refactoring in Hapi is manual.
+2. Endpoints must be created and tested manually.
+3. It restricts developers to using Hapi specific plugins and modules which are not compatible with other frameworks.
+
+#### Koa.js
+##### Advantages
+1. Koa has a small footprint hence making it very light weight.
+2. Koa has a good overall user experience.
+3. Koa uses try/catch which improves error handling.
+4. Koa has a cleaner and more readable asynchronous code.
+5. Koa does not use callbacks hence facilitating upstream and downstream flow of controls.
+
+##### Disadvantages
+1. Koa has a relatively small community.
+2. It is not compatible with middleware builds using Express style.
+3. Koa uses ES6 generators which are not compatible with middleware from other framework.
+
+#### Express.js
+##### Advantages
+1. Express has a large community.
+2. Express is simple, flexible, and scalable.
+3. Express is fully customizable.
+4. It is easy to integrate third party services and middleware.
+5. Express has a low learning curve.
+
+##### Disadvantages
+1. Express requires a lot of manual labor.
```

Comment:
In what way, does Express require a lot of manual labor. You should explain why this is true.